### ISSUE TITLE
VSTS Pester test extension: Parameters containing semicolons cannot be passed to scriptfolder property

### ISSUE THREAD

**Issue by bergmeister** [rfennell/vNextBuild#321]
### VSTS Extension you are using
VSTS Pester test extension

### Where are you running it?
VSTS, agent on-premise

### Version of Extension/Task
v7* (latest)

### Expected behaviour and actual behaviour
A build variable is defined as follows: `$files = 'foo.txt;bar.txt'` 
Then this variable is being passed into the extension (using a yaml build):
```
@{Path='.\tests\script.tests.ps1'; Parameters=@{Path='.\tests\script2.tests.ps1'; Parameters=@{otherparam='$(files)'}}
````
In the Pester test, only 'foo.txt' get passed to it despite the parameter value being surrounded by single quotes

**Comment by ChrisLGardner** [comment]
I'll have a look at this and see what we're doing to parse those parameters. It could be something involved in that which is dropping the extra values out, probably due to the semicolon I'd guess.

I'm guessing you want to keep the semicolon there for use within the tests?

**Comment by bergmeister** [comment]
@ChrisLGardner I created an hello-world example and sent you an invitation to your email address on your public GitHub profile.
- The YAML of the task is:
```yaml
- task: richardfennellBM.BM-VSTS-PesterRunner-Task.Pester-Task.Pester@7
  inputs:
    scriptFolder: '@{Path=''$(Build.SourcesDirectory)\fancypants.tests.ps1'; Parameters=@{Param1='foo.txt;bar.txt'}}'
    ForceUseOfPesterInTasks: true
````
![image](https://user-images.githubusercontent.com/9250262/39774321-64fe8b28-52f2-11e8-96ab-02136023ac88.png)

- The contents of `fancypants.tests.ps1` in the root directory is:
```powershell
Param($Param1)

Write-Verbose $Param1 -Verbose # Does not get displayed, another extension bug?
Write-Host $Param1 # RIP to a puppies
```
-When running it (e.g. on `Hosted VS2017`), then the log output is:
```
2018-05-08T17:52:44.5660970Z ##[section]Starting: Pester Test Runner
2018-05-08T17:52:44.5915473Z ==============================================================================
2018-05-08T17:52:44.5915674Z Task         : Pester Test Runner from Black Marble
2018-05-08T17:52:44.5915839Z Description  : Run Pester tests without the need to install Pester in with the PMModule folder or in the source repo (Using Pester 3.4.3 or 4.3.1)
2018-05-08T17:52:44.5916030Z Version      : 7.1.6
2018-05-08T17:52:44.5916136Z Author       : Richard Fennell
2018-05-08T17:52:44.5916274Z Help         : Version: 7.1.6. [More Information](https://github.com/rfennell/vNextBuild/wiki/Pester-Task/)
2018-05-08T17:52:44.5916440Z ==============================================================================
2018-05-08T17:52:44.6231634Z Preparing task execution handler.
2018-05-08T17:52:47.1289506Z Executing the powershell script: D:\a\_tasks\Pester_31ef0033-64e3-4c55-b888-f446541474a6\7.1.6\Pester.ps1
2018-05-08T17:52:55.6553466Z Executing all tests in 'D:\a\1\s\fancypants.tests.ps1'
2018-05-08T17:52:55.6554219Z 
2018-05-08T17:52:55.6554731Z 
2018-05-08T17:52:55.7883463Z 
2018-05-08T17:52:55.7883788Z 
2018-05-08T17:52:55.7904139Z Executing script D:\a\1\s\fancypants.tests.ps1
2018-05-08T17:52:55.7904413Z 
2018-05-08T17:52:55.7904551Z 
2018-05-08T17:52:55.7975350Z foo.txt
2018-05-08T17:52:55.7975633Z 
2018-05-08T17:52:55.7975814Z 
2018-05-08T17:52:55.9221612Z Tests completed in 0ms
2018-05-08T17:52:55.9222357Z 
2018-05-08T17:52:55.9222597Z 
2018-05-08T17:52:55.9443552Z Tests Passed: 0, 
2018-05-08T17:52:55.9466132Z Failed: 0, 
2018-05-08T17:52:55.9489112Z Skipped: 0, 
2018-05-08T17:52:55.9515510Z Pending: 0, 
2018-05-08T17:52:55.9539309Z Inconclusive: 0 
2018-05-08T17:52:55.9539646Z 
2018-05-08T17:52:55.9539854Z 
2018-05-08T17:52:57.9446885Z ##[section]Finishing: Pester Test Runner
````

- I would have expected to see `foo.txt;bar.txt` but it shows `foo.txt`

**Comment by ChrisLGardner** [comment]
I've managed to refactor the code we use to parse that hashtable so it's much nice and handles more random cases too. I'll try to get it merged in today and will probably release it as part of some other fixes I was working on (possibly as 8.x due to some breaking changes I was doing).

**Comment by ChrisLGardner** [comment]
Released as part of 7.2.1

**Comment by bergmeister** [comment]
@ChrisLGardner Thanks. I can confirm this is fixed now
